### PR TITLE
feat: use data-store for subdomain data

### PR DESCRIPTION
**Pull request by colbr**
## 1. Pull request checklist

n/a

## 2. PR type
 

Feature

## 3. What is the old behaviour?

NFTs zApp retrieved subdomain data through Infura.

## 4. What is the new behaviour?

NFTs zApp retrieves subdomain data through the data store.

## 5. Other information

We will still need to switch some pages/components over to using the data store - this PR mostly focuses on the subdomain table.

### CODE REVIEW

File: src/features/view-subdomains/SubdomainTable/useInfiniteSubdomains.ts [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+			const currentIndex = pageParam * DEFAULT_RESULTS_PER_PAGE;
+
+			console.log(parentDomainId);
+			console.log('currin', currentIndex, 'pageparam', pageParam);
+			sdk
+				.getSubdomainsById(parentDomainId, true, 24, 0, DEFAULT_SORT_OPTIONS)
+				.then((d) => console.log('24', d.map((d) => d.name).sort()));
+			sdk
+				.getSubdomainsById(parentDomainId, true, 24, 24, DEFAULT_SORT_OPTIONS)
+				.then((d) => console.log('48', d.map((d) => d.name).sort()));
```
Comment (colbr) [inline src/features/view-subdomains/SubdomainTable/useInfiniteSubdomains.ts:53]: Apologies forgot to remove this, ignore it I'll remove it tomorrow

File: src/features/view-subdomains/SubdomainTable/SubdomainTable.tsx [hunk 2]
```diff
@@ -1,66 +1,223 @@
...
+		!isLoadingInitialSubdomains &&
+		!isLoadingNextSubdomains &&
+		!isFetchingSearch &&
+		!isRefetching &&
+		activeQuery &&
+		!isIdle &&
+		subdomains.length === 0;
+
+	return (
+		<div style={{ position: 'relative' }}>
```
Comment (domw30) [inline src/features/view-subdomains/SubdomainTable/SubdomainTable.tsx:72]: nit: inline style could be added to the scss file

File: src/features/view-subdomains/SubdomainTable/SubdomainTable.tsx [hunk 3]
```diff
@@ -1,66 +1,223 @@
...
+			)}
+			{isLoadingInitialSubdomains && (
+				<TableStatusMessage
+					status={TableStatus.LOADING}
+					message={'Loading subdomains'}
+				/>
+			)}
+			{isEmpty && <TableStatusMessage status={TableStatus.EMPTY} />}
+			{isLoadingNextSubdomains && (
+				<TableStatusMessage
```
Comment (domw30) [inline src/features/view-subdomains/SubdomainTable/SubdomainTable.tsx:98]: nice addition 💪

File: src/features/view-subdomains/SubdomainTableRow/SubdomainTableRow.tsx [hunk 4]
```diff
@@ -65,25 +60,26 @@ export const SubdomainTableRow: FC<SubdomainTableRowProps> = ({
 						<span>0://{zna}</span>
 					</div>
 				</div>
-			</TableData>
-			<TableData alignment={'right'} className={styles.Metrics}>
+			</Cell>
+			<Cell alignment={'right'} className={styles.Metrics}>
 				<SkeletonText
 					asyncText={{
 						text: volume,
```
Comment (domw30) [inline src/features/view-subdomains/SubdomainTableRow/SubdomainTableRow.tsx:67]: Weird I thought I fixed these up...same as the card -

Whilst we're here could we add `errorText: '-',` below line 68: 

Screenshot 2023-03-01 at 18 34 30

File: src/features/view-subdomains/SubdomainTableCard/SubdomainTableCard.tsx [hunk 5]
```diff
@@ -60,13 +55,10 @@ export const SubdomainTableCard: FC<SubdomainTableCardProps> = ({
 		}
 	});
 
-	const metric = buyNowPrice?.price
-		? `${formatEthers(buyNowPrice?.price.toString())} ${paymentTokenSymbol}`
+	const metric = buyNowPrice
```
Comment (domw30) [inline src/features/view-subdomains/SubdomainTableCard/SubdomainTableCard.tsx:58]: Whilst we're here could we add `errorText: '-',` below line 96: 

Screenshot 2023-03-01 at 15 56 44

File: src/features/view-subdomains/SubdomainTable/SubdomainTable.tsx [hunk 6]
```diff
@@ -1,66 +1,223 @@
...
-			data={subdomains}
-			itemKey="id"
-			columns={COLUMNS}
-			rowComponent={Row}
-			gridComponent={Card}
-			searchKey={{ key: 'name', name: 'zna' }}
-			isLoading={isLoadingSubdomains}
-		/>
+		<div
+			style={{
```
Comment (domw30) [inline src/features/view-subdomains/SubdomainTable/SubdomainTable.tsx:216]: nit: inline styles to scss file

File: src/features/view-subdomains/SubdomainTable/SubdomainTable.tsx [hunk 7]
```diff
@@ -1,66 +1,223 @@
...
+				position: 'absolute',
+				bottom: '75vh',
+			}}
+			ref={triggerRef}
+		></div>
```
Comment (domw30) [inline src/features/view-subdomains/SubdomainTable/SubdomainTable.tsx:221]: nit: drop div a line down

File: src/features/view-subdomains/SubdomainTable/useInfiniteSubdomains.ts [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+	 */
+	const fetchSubdomains = useCallback(
+		async ({ pageParam = 0 }) => {
+			// Number of results already loaded, used to determine where to start the next query
+			const currentIndex = pageParam * DEFAULT_RESULTS_PER_PAGE;
+
+			console.log(parentDomainId);
+			console.log('currin', currentIndex, 'pageparam', pageParam);
+			sdk
+				.getSubdomainsById(parentDomainId, true, 24, 0, DEFAULT_SORT_OPTIONS)
```
Comment (domw30) [inline src/features/view-subdomains/SubdomainTable/useInfiniteSubdomains.ts:49]: Would it make sense to name the hardcoded values e.g. `const resultsPerPage = 24` or something.

Oh not sure if this is to be removed as per comment below ...

File: src/features/view-subdomains/SubdomainTable/useInfiniteSubdomains.ts [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+				.then((d) => console.log('24', d.map((d) => d.name).sort()));
+			sdk
+				.getSubdomainsById(parentDomainId, true, 24, 24, DEFAULT_SORT_OPTIONS)
+				.then((d) => console.log('48', d.map((d) => d.name).sort()));
+
+			const data = await sdk.getSubdomainsById(
+				parentDomainId,
+				true,
```
Comment (domw30) [inline src/features/view-subdomains/SubdomainTable/useInfiniteSubdomains.ts:57]: Might be worth adding a global constant for `useDataStoreApi` / `isUsingDataStoreApi` and set it as true. Not sure if that would be overkill though

File: src/features/view-subdomains/SubdomainTable/useInfiniteSubdomains.ts [hunk 10]
```diff
@@ -0,0 +1,109 @@
...
+		[parentDomainId, sdk],
+	);
+
+	// Note: assigning query to variable so we can destructure it in hook return
+	const query = useInfiniteQuery({
+		queryKey,
+		queryFn: fetchSubdomains,
+		getNextPageParam: (lastPage, pages) => {
+			// Doesn't look like the API supports a way to determine if there are more results
+			// so we just check if the last page is full, and if so assume there are more
```
Comment (domw30) [inline src/features/view-subdomains/SubdomainTable/useInfiniteSubdomains.ts:78]: 👍

File: src/features/view-subdomains/useSubdomainTableItem.ts [hunk 11]
```diff
@@ -1,39 +1,68 @@
...
-	const { data: domain } = useDomainData(domainId);
 	const { data: metrics, isLoading: isLoadingMetrics } =
 		useDomainMetrics(domainId);
-	const { data: buyNowPrice, isLoading: isLoadingBuyNowPrice } =
-		useBuyNowListing(domainId);
 	const { data: metadata, isLoading: isLoadingMetadata } =
 		useDomainMetadata(domainId);
 	const { data: paymentToken } = usePaymentToken(parentZna);
 
+	const isOwnedByUser = account && subdomain?.owner === account;
```
Comment (domw30) [inline src/features/view-subdomains/useSubdomainTableItem.ts:48]: We'll need to re-add the `.toLowerCase()` here to match the strings and determine if the user owns the domain. I see the Buy/Bid button if I own the domain at the moment

File: src/features/view-subdomains/SubdomainTableRow/SubdomainTableRow.tsx [hunk 12]
```diff
@@ -65,25 +60,26 @@ export const SubdomainTableRow: FC<SubdomainTableRowProps> = ({
...
 					}}
 				/>
-			</TableData>
+			</Cell>
 
-			{!isOwnedByUser && (
-				<TableData alignment={'right'} className={styles.Button}>
-					{buyNowPrice ? (
+			<Cell alignment={'right'} className={styles.Button}>
+				{!isOwnedByUser &&
```
Comment (domw30) [inline src/features/view-subdomains/SubdomainTableRow/SubdomainTableRow.tsx:74]: Assuming this switch is because we still want to render the empty cell?

**Review by domw30 — CHANGES_REQUESTED** [review]
Looks awesome mate. There's one requested change in there (`toLowerCase`) as it looks broken at the moment. Lmk if I've missed something with that. 🚀

File: src/features/view-subdomains/SubdomainTable/useInfiniteSubdomains.ts (hunk 9 above)
Comment (colbr) [reply src/features/view-subdomains/SubdomainTable/useInfiniteSubdomains.ts:57]: True, I'll move this to a global variable for now, and we can move it to an environment variable or a feature flag next step

File: src/features/view-subdomains/SubdomainTable/useInfiniteSubdomains.ts (hunk 8 above)
Comment (colbr) [reply src/features/view-subdomains/SubdomainTable/useInfiniteSubdomains.ts:49]: Yep removed this one!

File: src/features/view-subdomains/SubdomainTable/SubdomainTable.tsx (hunk 6 above)
Comment (colbr) [reply src/features/view-subdomains/SubdomainTable/SubdomainTable.tsx:216]: Good call

File: src/features/view-subdomains/SubdomainTableRow/SubdomainTableRow.tsx (hunk 12 above)
Comment (colbr) [reply src/features/view-subdomains/SubdomainTableRow/SubdomainTableRow.tsx:74]: Correct!

**Review by domw30 — APPROVED** [review]
LGTM ! 💯

**Pull request merged**